src/app-layer-htp-file.c

```diff
@@ -248,7 +260,48 @@ int HTPFileSetRange(HtpState *s, bstr *rawvalue)
         SCLogDebug("negative end in range");
         SCReturnInt(-3);
     }
-    int retval = FileSetRange(files, crparsed.start, crparsed.end);
+
+    flags = FileFlowToFlags(s->f, STREAM_TOCLIENT);
+    if ((s->flags & HTP_FLAG_STORE_FILES_TS) ||
+        ((s->flags & HTP_FLAG_STORE_FILES_TX_TS) && txid == s->store_tx_id)) {
+        flags |= FILE_STORE;
+        flags &= ~FILE_NOSTORE;
+    } else if (!(flags & FILE_STORE) && (s->f->file_flags & FLOWFILE_NO_STORE_TC)) {
+        flags |= FILE_NOSTORE;
+    }
+
+    //TODO handle unordered ranges
+    if (crparsed.start == 0 && crparsed.end < crparsed.size) {
+        // open another file for the whole ranges
+        if (s->file_range_id != 0) {
+            // TODO handle parallel range downloads
+            if (FileCloseFileById(files, s->file_range_id, NULL, 0, FILE_TRUNCATED) != 0) {
+                SCLogDebug("close file for range failed");
+            }
+        }
+        //TODO use a flag ?
+        s->file_range_id = ~s->file_track_id;
```

Comment:
`file_track_ids` are integer values ... not sure why you're doing this?

Reply:
I want some ID that will not be overrun by regular/usual file ids
As the comment states, it may be better to have `s->file_range_id = FLAG_FILE_RANGES | s->file_track_id;` with FLAG_FILE_RANGES = 0x80000000

Reply:
That might be clearer.

Reply:
ok